File: user-service/src/main/java/rs/raf/user_service/domain/dto/CreateVerificationRequestDto.java
package rs.raf.user_service.domain.dto;

import lombok.*;
import rs.raf.user_service.domain.enums.VerificationType;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateVerificationRequestDto {
    @NotNull
    private Long userId;
    @NotNull
    private Long targetId;
    @NotNull
    private VerificationType verificationType;
    @NotNull
    @NotBlank
    private String details;
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/dto/verification/CreateVerificationRequestDto.kt
package pack.userservicekotlin.domain.dto.verification

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import pack.userservicekotlin.domain.enums.VerificationType

data class CreateVerificationRequestDto(
    @NotNull
    val userId: Long? = null,
    @NotNull
    val targetId: Long? = null,
    @NotNull
    val verificationType: VerificationType? = null,
    @NotNull
    @NotBlank
    val details: String? = null,
)
